### PR TITLE
clock: esp32: unify clock control for all espressif socs

### LINKED ISSUE #39569: [ESP32] crash when trying to set a low cpu clock frequency
Hello,

I would like to change the default cpu frequency of the esp32 from 240M to a lower value to reduce power.
I don't know if this is supported, but I tried to modify the esp32.dts to set an other frequency, rebuild,flash... and after the bootloader finished, I see a few garbage characters in the serial output and then nothing.

I suspect that either I didn't change the clock properly or there is a bug somewhere in the DRAM or peripheral clock computation

**To Reproduce**
1. Build the samples/bluetooth/peripheral_ht (for example)
2. modify boards/xtensa/esp32/esp32.dts and replace ESP32_CLK_CPU_240M by any other supported value declared in include/dt-bindings/clock/esp32_clock.h, for example ESP32_CLK_CPU_160M
3. rebuild and flash

**Expected behavior**
Changing the cpu clock should work and application should run properly

**Impact**
I need to got sub 100mA because it's a low power project

**Logs and console output**
clk_drv:0x00,q_drv:0x00,d_drv:0x00,cs0_drv:0x00,hd_drv:0x00,wp_drv:0x00                                                zmode:DIO, clock div:2                                                                                                  z
load:0x3fff0030,len:7264
load:0x40078000,len:14292
ho 0 tail 12 room 4
load:0x40080400,len:3688
entry 0x40080678
I (28) boot: ESP-IDF release/v4.3-94-g2308568ea 2nd stage bootloader
I (28) boot: compile time 22:00:32
I (28) boot: chip revision: 1
I (32) boot_comm: chip revision: 1, min. bootloader chip revision: 0
I (39) boot.esp32: SPI Speed      : 40MHz
I (44) boot.esp32: SPI Mode       : DIO
I (49) boot.esp32: SPI Flash Size : 4MB
I (53) boot: Enabling RNG early entropy source...
I (59) boot: Partition Table:
I (62) boot: ## Label            Usage          Type ST Offset   Length
I (69) boot:  0 nvs              WiFi data        01 02 00009000 00006000
I (77) boot:  1 phy_init         RF data          01 01 0000f000 00001000
I (84) boot:  2 factory          factory app      00 00 00010000 00100000
I (92) boot: End of partition table
I (96) boot_comm: chip revision: 1, min. application chip revision: 0
I (103) esp_image: segment 0: paddr=00010020 vaddr=3f400020 size=0449ch ( 17564) map
I (118) esp_image: segment 1: paddr=000144c4 vaddr=3ffbdb5c size=0121ch (  4636) load
I (122) esp_image: segment 2: paddr=000156e8 vaddr=3ffbed78 size=00174h (   372) load
I (129) esp_image: segment 3: paddr=00015864 vaddr=40080000 size=00400h (  1024) load
I (138) esp_image: segment 4: paddr=00015c6c vaddr=40080400 size=0a3ach ( 41900) load
I (164) esp_image: segment 5: paddr=00020020 vaddr=400d0020 size=25560h (152928) map
I (221) esp_image: segment 6: paddr=00045588 vaddr=4008a7ac size=05f28h ( 24360) load
I (241) boot: Loaded app from partition at offset 0x10000
I (241) boot: Disabling RNG early entropy source...                                                                    t쭗�~Ѩ٨Fxp�      izѨPX

**Environment:**
 - OS: Linux

### PR DESCRIPTION
**Pull request by sylvioalves**
This joins all clock control handling to same source
by using hal clock functions. It also brings ESP32C3
clock support.

Closes #39569

Signed-off-by: Sylvio Alves <sylvio.alves@espressif.com>

### CODE REVIEW

File: drivers/clock_control/Kconfig.esp32 [hunk 1]
```diff
@@ -5,6 +5,6 @@
 
 config CLOCK_CONTROL_ESP32
 	bool "ESP32 Clock driver"
-	depends on SOC_ESP32
+	depends on SOC_ESP32 || SOC_ESP32S2 || SOC_ESP32C3
```
Comment (sylvioalves) [inline drivers/clock_control/Kconfig.esp32:8]: I'll work on this in another PR to have some `SOC_ESPRESSIF_FAMILY`

**Review by uLipe — APPROVED** [review]

**Review by glaubermaroto — APPROVED** [review]

**Comment by sylvioalves** [comment]
Rebased only.

**Review by uLipe — APPROVED** [review]

**Comment by cfriedt** [comment]
I don't understand why @uLipe's review is not green. Likely something to do with recently GH permission changes. @nashif - any ideas why?

**Review by cfriedt — APPROVED** [review]

**Comment by sylvioalves** [comment]
> I don't understand why @uLipe's review is not green. Likely something to do with recently GH permission changes. @nashif - any ideas why?

@uLipe and I have been talking about it. All his approvals aren't green..

**Pull request merged**